### ISSUE TITLE
win32: strdup vs _strdup causing MSVC runtime heap check debug asserts and crashes

### ISSUE THREAD

**Issue by TTimo** [zeromq/czmq#1041]
context: MSVC 2010, vc100 toolset, all libraries using Debug Runtime DLL CRT

I ran into a crash that would only happen in debug builds. It is preceded by an assertion failure from MSVC heap checking when freeing the filename in in zhashx_destroy.

I noticed that czmq on win32 is somehow using strdup .. I don't understand how it does this, as win32 is only supposed to have _strdup (see http://stackoverflow.com/questions/7582394/strdup-or-strdup for some discussions).

In zhashx_load, I replaced 

``` c
    char *filename_copy = strdup (filename);
```

with

``` c
    char *filename_copy = _strdup (filename);
```

And the crash went away. So I proceeded to make my czmq copy use _strdup everywhere on Windows. That fixed some other similar crashes (PLAIN passwords for auth v2 were doing the same thing).

TL;DR - I think czmq should be using _strdup on win32. It certainly fixes a crash for me.

**Comment by Asmod4n** [comment]
This should be placed into czmq_prelude.h, somewhere after https://github.com/zeromq/czmq/blob/master/include/czmq_prelude.h#L425 perhaps.

**Comment by evoskuil** [comment]
Are you statically linking the c runtime or are any of your dependencies?
